guides/novu-fcm-web.mdx

```diff
@@ -0,0 +1,219 @@
+```jsx
+importScripts("https://www.gstatic.com/firebasejs/8.10.0/firebase-app.js");
+importScripts(
+    "https://www.gstatic.com/firebasejs/8.10.0/firebase-messaging.js"
+);
+
+const firebaseConfig = {
+    apiKey: import.meta.env.VITE_FIREBASE_API_KEY,
+    authDomain: import.meta.env.VITE_FIREBASE_AUTH_DOMAIN,
+    projectId: import.meta.env.VITE_FIREBASE_PROJECT_ID,
+    storageBucket: import.meta.env.VITE_FIREBASE_STORAGE_BUCKET,
+    messagingSenderId: import.meta.env.VITE_FIREBASE_MESSAGING_SENDER_ID,
+    appId: import.meta.env.VITE_FIREBASE_APP_ID,
+};
+
+firebase.initializeApp(firebaseConfig);
+const messaging = firebase.messaging();
+messaging.onBackgroundMessage((payload) => {
+    console.log(
+        "notif incoming",
+        payload
+    );
+    const notificationTitle = JSON.parse(JSON.parse(payload.notification)).title
+    // payload.notification.title;
+    const notificationOptions = {
+        body: payload.notification.body,
+        icon: payload.notification.image,
+    };
+
+    self.registration.showNotification(notificationTitle, notificationOptions);
+});
+
+
+```
+
+## Adding Novu to the backend
+In your app's backend, add Novu using the following command:
+```bash
+npm install @novu/node
+```
+Now, create a route that you want to hit when called from the front end. In our app, this is the route:
+```jsx
+import express from "express";
+import { createNotif } from '../controller/notif.js'
+
+const router = express.Router();
+
+router.post('/create', createNotif)
+
+export default router;
+```
+
+Now, we need a controller function to handle the logic for what is to be sent in the trigger function’s payload. In our case, this is the controller function:
+```jsx
+import { inAppNotification } from "../novu/novu.js";
+
+export const createNotif = async (req, res) => {
+
+    const { title, body } = req.body
+    try {
+        await inAppNotification(title, body);
+        res.status(201).json({ message: 'success', title: title, body: body });
+    } catch (error) {
+        res.status(409).json({ message: error.message });
+    }
+}
+```
+
+To make it modular, we’ll keep the trigger code in a separate function in a separate file (`novu.js`, in our case) and the trigger function is getting called in the controller function above by the name `createNotif`.
+
+If you’re following the guide, you should already have the trigger function. But before we can add it to our app, we need one key thing - **Subscribers.**
+
+Subscribers are entities to which the notifications are sent. You can see a list of subscribers in the [Novu dashboard](https://web.novu.co/subscribers) as well.
+
+In our app, we’ll create a subscriber in Node.js as we’re writing our backend in Node.js, but we also have backend [SDKs](/sdks/introduction) (Node.js, PHP, .NET, Go, Ruby, Python, and Kotlin) to choose from. The recommended way to create a subscriber in NodeJS is as follows:
```

Comment:
> In our app, we’ll create a subscriber in Node.js as we’re writing our backend in Node.js, but we also have backend [SDKs](/sdks/introduction) (Node.js, PHP, .NET, Go, Ruby, Python, and Kotlin) to choose from.

We have repetition here. This can be rephrased to:

```
We'll create a subscriber in our app. Our backend will be written in Node.js, but we also have backend [SDKs in PHP, .NET, Go, Ruby, Python, and Kotlin](/sdks/introduction).
```